src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -13,6 +13,7 @@
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
```

Comment:
Sorry, I still don't completely understand what a session is so can you please explain how you're testing this.

Reply:
Since the doPost calls getSession() and then getAttribute() on the session it gets from the first method to get the name of the logged in user(if any), I just made a mockSession and used the when() method to simulate the calls.

Reply:
I see, thanks.